### PR TITLE
M3-5208: Fix Domains table header for large accounts

### PR DESCRIPTION
**Pull request by tiffwong**
## Description

Tables using the `APIPaginatedTable` component did not have the right styling and are missing the last column in the header since the props weren't being passed down.

## How to test

Login to `prod-test-001` and go to `/domains`

### CODE REVIEW

**Review by HanaXu — COMMENTED** [review]
Not sure how big of an effort it would be, but can we also standardize the domains header styling to match the other headers?

image

image

image

image

image

**Review by DevDW — COMMENTED** [review]
In addition to the styling tweaks for dark mode, the grouping by tag functionality doesn't seem to be working here

**Comment by tiffwong** [comment]
@HanaXu the table should match now, lmk if I missed anything!

@DevDW we discussed this but for posterity's sake, turns out Group by Tag is disabled for large accounts. See [this PR](https://github.com/linode/manager/pull/6653#pullrequestreview-451921762) for more details on API pagination

**Review by HanaXu — COMMENTED** [review]
Looking a lot better! Just have some comments

The Domains Last Modified Header is looking a bit wide and I don't think the date format will be longer than it is currently?
image

And for tables in general, the table header hover states are kind of inconsistent. (you don't have to address now, just something to note).

In Linodes and StackScripts, the whole header is clickable and the hover state is a dark blue:

image

In Volumes, NodeBalancers, Firewalls, Images and Domains, you need to hover over the actual word to be able to filter.

In Kubernetes and Object Storage, the whole header is clickable but the hover state is a different shade of blue:

image

I do think we should have the whole header clickable instead of just the text tho. We should also figure out which blue hover state we want to stick with.

File: packages/manager/src/components/EntityTable/EntityTableHeader.tsx [hunk 1]
```diff
@@ -205,6 +196,7 @@ export const GroupByTagToggle: React.FC<GroupByTagToggleProps> = React.memo(
             onClick={toggleGroupByTag}
             disableRipple
             className={classes.toggleButton}
+            disabled={isLargeAccount}
```
Comment (DevDW) [inline packages/manager/src/components/EntityTable/EntityTableHeader.tsx:199]: might be worth adding a comment explaining why this is the case

**Review by DevDW — APPROVED** [review]

File: packages/manager/src/features/Domains/DomainsLanding.tsx [hunk 2]
```diff
@@ -126,67 +75,92 @@ interface Props {
...
-    removeDialogLoading: false,
-    selectedDomainLabel: '',
-    disableDialogOpen: false,
-  };
-
-  static docs: Linode.Doc[] = [Domains];
+const getHeaders = (
+  matchesXsDown: boolean,
+  matchesSmDown: boolean,
+  matchesMdDown: boolean
```
Comment (tiffwong) [inline packages/manager/src/features/Domains/DomainsLanding.tsx:88]: Ugly but couldn't figure out a better way to do this

File: packages/manager/src/features/Domains/DomainsLanding.tsx [hunk 3]
```diff
@@ -42,79 +39,31 @@ import {
...
-    },
-  });
-
-interface State {
-  importDrawerOpen: boolean;
-  importDrawerSubmitting: false;
-  importDrawerErrors?: APIError[];
-  remote_nameserver: string;
-  createDrawerOpen: boolean;
-  createDrawerMode: 'clone' | 'create';
```
Comment (tiffwong) [inline packages/manager/src/features/Domains/DomainsLanding.tsx:107]: Removed these props since they're no longer being used

**Comment by tiffwong** [comment]
@HanaXu great catch on the header styles! I'll address those in a follow up pr 😂

**Review by HanaXu — APPROVED** [review]
LGTM!

**Pull request merged**